### ISSUE TITLE
Flush spotify streams before unmuting

### ISSUE THREAD

**Issue by mahkoh** [mahkoh/spotblock#1]
When the streams are unmuted they often still contain a bit of ad because they are buffered.

**Comment by ghost** [comment]
Yes I can confirm this behaviour, it is just below a second. Thanks for the good work!

**Comment by mahkoh** [comment]
I don't think this can be fixed. The real solution is for spotify to become latency aware. You can see the same effects in the UI when it already shows the next song even though the old one is still playing.

**Comment by ghost** [comment]
Why not just adding half a sec of an offset value?

**Comment by mahkoh** [comment]
Unfortunately the latency is not consistent.

**Comment by gCurse** [comment]
What about adding an unmute delay of about 5 seconds and prior to  unmuting encourage spotify to jump to the beginning of the current song?

**Comment by mahkoh** [comment]
Ideally we would seek to 0 once the ad ends. This would flush the buffer. However, spotify does not support seeking via MPRIS.

**Comment by gCurse** [comment]
But spotify supports "Previous" which seeks to 0 if the song is playing for about 3 (or more) seconds.
I now use [this script](https://gist.github.com/gCurse/66e00cde4aea83d1589df2edd705904f) (see lines 56-57) and it works ~~~seamlessly~~~ fine for me.

**Comment by mahkoh** [comment]
I'd hardly call this working seamlessly:

```python
            if advert:
                advert = False
                time.sleep(6.0)
                player.Previous()

            pulsesink_mute(0)
```

But since the interval is only 3 seconds I might think about implementing this properly.

**Comment by gCurse** [comment]
How would you ever "implement this properly"?
As you stated: "Unfortunately the latency is not consistent." Right!
So every user has to figure out his personal `max(latency)`. In my case the six seconds delay works.
But OK, I might be wrong. Awaiting your proof.

BTW: [politeness: showing good manners toward others, as in behaviour, speech, etc.](https://www.merriam-webster.com/thesaurus/politeness)
*muted*

**Comment by ghost** [comment]
If it helps, here's how to do prev for spotify:
```
dbus-send --print-reply --dest=org.mpris.MediaPlayer2.spotify /org/mpris/MediaPlayer2 org.mpris.MediaPlayer2.Player.Prev
```

**Comment by mxmilkiib** [comment]
Presumably, in this case, at the point of the notification, the audio buffer power level is at non 0 (playing the tail end of the ad). The audio buffer level presumably then goes to 0 output amplitude for at least a sample at the end of the advert, no? Could detecting that 0 volume sample be used to trigger unmuting the audio?
